### PR TITLE
[WIP]: Tree-shakable providers documentation

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

No docs existing for new-feature tree-shakable providers.
Issue Number: N/A

## What is the new behavior?
Added tree-shakable provider documentation with a recommendation to use it as the default method.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by chembu** [comment]
@jenniferfell , @alxhub : Take a look at the tree-shakable providers doc for V6.

**Comment by trotyl** [comment]
Should be able to close #21913 as well

File: aio/content/guide/styleguide.md [hunk 1]
```diff
@@ -4663,7 +4663,7 @@ Compare with the less preferred `host` metadata alternative.
 
 
 
-**Do** provide services to the Angular injector at the top-most component where they will be shared.
+**Do** provide services with the app root injector in the `@Injectable` decorator of the service.
```
Comment (jenniferfell) [inline aio/content/guide/styleguide.md]: Trying to resolve the plural - singular shift. Maybe "Do provide a service with the app root injector in the @Injectable decorator of the service" ?

File: aio/content/guide/styleguide.md [hunk 2]
```diff
@@ -4685,8 +4685,8 @@ Compare with the less preferred `host` metadata alternative.
 
 
 
-**Why?** When providing the service to a top level component,
-that instance is shared and available to all child components of that top level component.
+**Why?** When providing the service to a root injector,
+that instance is shared and available in every class that needs the service. And provding it in the `@Injectable` decorator allows Angular compiler to perform tree-shaking on services that are not used in an application.
```
Comment (jenniferfell) [inline aio/content/guide/styleguide.md]: Recommend: "...that instance of the service is shared.."

Comment (jenniferfell) [inline aio/content/guide/styleguide.md]: Last sentence could use some editing. 
- Rewrite to not start sentence with And
- Providing is missing an i
- "It" could be more clear
- "an application"  ->  "the app"
Perhaps "Providing that instance of the service in the @Injectable decorator allows the Angular compiler to perform tree-shaking on services that are not used it the app."

File: aio/content/guide/providers.md [hunk 3]
```diff
@@ -10,32 +10,40 @@ see the <live-example></live-example>.
 <hr>
 
 ## Create a service
-You can provide services to your app by using the `providers` array in an NgModule.
-Consider the default app generated by the CLI. In order to add a user service to it,
-you can generate one by entering the following command in the terminal window:
+Assuming you already have a CLI generated app, create a service using the following CLI command (replace User with the name of your service) in the root project directory:
```
Comment (jenniferfell) [inline aio/content/guide/providers.md]: Editorial fine-tuning: 
- "If you have an app generated that was generated with the CLI..."
Or 
- "If you have a CLI-generated app, ..."

Comment (jenniferfell) [inline aio/content/guide/providers.md]: I recommend pulling out the parenthetical. "...create a service using the following CLI command in the root project directory. Replace User with the name of your service."

Comment (jenniferfell) [inline aio/content/guide/providers.md]: What if I don't have a CLI-generated app?

**Review by jenniferfell — CHANGES_REQUESTED** [review]
Partial review complete

File: aio/content/guide/providers.md [hunk 4]
```diff
@@ -10,32 +10,62 @@ see the <live-example></live-example>.
...
-app's injector. Update `app.module.ts` by importing it with your other import statements at the top
-of the file and adding it to the `providers` array:
-
-<code-example path="providers/src/app/app.module.ts" title="src/app/app.module.ts" linenums="false">
+  
+This creates a service called `UserService`, which will look similar to the code snippet shown below: 
+
+<code-example>
+
+@Injectable({
```
Comment (alxhub) [inline aio/content/guide/providers.md]: Example: user.service.0.ts

File: aio/content/guide/providers.md [hunk 5]
```diff
@@ -10,32 +10,62 @@ see the <live-example></live-example>.
...
+When you add a service provider to the root application injector, it’s available throughout the app. Additionally, these providers are also available to all the classes in the app as long they have the lookup token.
+
+Unless there is a special case, you should always provide your service in the root injector.
+
+## providedIn and NgModules
+
+It's also possible to specify that a service should be provided in a particular `@NgModule`. For example, if you don't want the `UserService` to be available to applications unless they import a `UserModule` you've created, you can specify that the service should be provided in the module:
 
+<code-example>
+@Injectable({
```
Comment (alxhub) [inline aio/content/guide/providers.md]: Example: user.service.1.ts

File: aio/content/guide/providers.md [hunk 6]
```diff
@@ -10,32 +10,62 @@ see the <live-example></live-example>.
...
+    providedIn: UserModule, 
+    }) 
+export class UserService {}
+</code-example>
+
+This is the preferred way to provide a service in a module, and enables tree-shaking of the service if nothing injects it. If it's not possible to specify in the service which module should provide it, you can also declare a provider for the service within the module:
+
+<code-example>
+@NgModule({ 
+
```
Comment (alxhub) [inline aio/content/guide/providers.md]: Example: user.module.ts

File: aio/content/guide/styleguide.md [hunk 7]
```diff
@@ -4715,17 +4715,18 @@ that instance is shared and available to all child components of that top level
 
 
 
-<code-tabs>
-
-  <code-pane title="app/app.component.ts" path="styleguide/src/07-03/app/app.component.ts">
+<code-example>
```
Comment (alxhub) [inline aio/content/guide/styleguide.md]: Example: hero.service.ts

File: aio/content/guide/dependency-injection-in-action.md [hunk 8]
```diff
@@ -11,7 +11,11 @@ of the code in this cookbook.
 {@a app-wide-dependencies}
 
 ## Application-wide dependencies
-Register providers for dependencies used throughout the application in the root application component, `AppComponent`.
+
+From V6 onwards, providers for application-wide dependencues should be defined by specifying `providedIn: 'root'` on their `@Injectable()` decorator (in the case of services) or at `InjectionToken` construction (in the case of tokens being provided).
```
Comment (alfaproject) [inline aio/content/guide/dependency-injection-in-action.md]: dependencies*

**Review by alfaproject — CHANGES_REQUESTED** [review]

**Comment by chembu** [comment]
New PR: https://github.com/angular/angular/pull/23027

**Comment by jenniferfell** [comment]
@alfaproject Thanks. Good catch!

**Comment by jenniferfell** [comment]
Moved work to https://github.com/angular/angular/pull/23027

**Pull request closed without merging**